src/ClientManager/Travely.ClientManager.Repository/Repository/BaseRepository.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Travely.ClientManager.Repository.Abstraction;
+using Travely.ClientManager.Repository.Entity;
+
+namespace Travely.ClientManager.Repository.Repository
+{
+    public class BaseRepository<T> : IBaseRepository<T> where T : BaseEntity
+    {
+        protected TuristDbContext _dbContext;
+        public BaseRepository(TuristDbContext dbContext)
+        {
+            _dbContext = dbContext;
+
+        }
+
+        #region Base CRUD Operation
+
+        public void Add(T entity)
+        {
+            _dbContext.Set<T>().Add(entity);
+        }
+
+        public void Delete(T entity)
+        {
+            EntityEntry<T> dbEntityEntry = _dbContext.Entry(entity);
+
+            _dbContext.Set<T>().Attach(entity);
+
+            dbEntityEntry.State = EntityState.Deleted;
+        }
+
+        public IQueryable<T> Get(Expression<Func<T, bool>> predicate, params string[] includes)
+        {
+            IQueryable<T> set = _dbContext.Set<T>();
+
+            for (int i = 0; i < includes.Length; i++)
+            {
+                set = set.Include(includes[i]);
+            }
+
+            if (predicate == null)
+            {
+                return set;
+            }
+
+            return set.Where(predicate);
+        }
+
+        public IQueryable<T> Get(params string[] includes)
```

Comment:
You have code duplication in this 2 methods `IQueryable<T> Get(Expression<Func<T, bool>> predicate, params string[] includes)` and `IQueryable<T> Get(params string[] includes)`